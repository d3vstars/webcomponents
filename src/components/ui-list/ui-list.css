.ui-list-table-content {
  max-width: 100%;
  margin: 20px 0px;
}

.ui-list-table {
  font-family: 'Helvetica Neue', sans-serif;
  background-color: transparent;
  border-collapse: collapse;
  border-spacing: 0;
  min-width: calc(100% - 2px);
}

.ui-list-table thead {
  background-color: #4e71c0;
  font-size: 12px;
}

.ui-list-table thead th {
  border-bottom: none;
  cursor: pointer;
  font-weight: normal;
  text-align: left;
  padding: 10px 15px;
  background-color: #999999;
  color: #ffffff;
  font-size: 13px;
  white-space: nowrap;
  background-color: #4e71c0;
}

.ui-list-table thead th p {
  margin: 0;
  float: left;
}

.ui-list-table thead th i {
  float: left;
  margin-left: 10px;
}

.ui-list-table thead th .withoutOrderBy {
  width: 7px;
  height: 14px;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxzdmcKICAgeG1sbnM6ZGM9Imh0dHA6Ly9wdXJsLm9yZy9kYy9lbGVtZW50cy8xLjEvIgogICB4bWxuczpjYz0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjIgogICB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiCiAgIHhtbG5zOnN2Zz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciCiAgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgeG1sbnM6c29kaXBvZGk9Imh0dHA6Ly9zb2RpcG9kaS5zb3VyY2Vmb3JnZS5uZXQvRFREL3NvZGlwb2RpLTAuZHRkIgogICB4bWxuczppbmtzY2FwZT0iaHR0cDovL3d3dy5pbmtzY2FwZS5vcmcvbmFtZXNwYWNlcy9pbmtzY2FwZSIKICAgd2lkdGg9IjgiCiAgIGhlaWdodD0iMTIuNSIKICAgdmlld0JveD0iMCAwIDggMTIuNSIKICAgZmlsbD0ibm9uZSIKICAgdmVyc2lvbj0iMS4xIgogICBpZD0ic3ZnODM1IgogICBzb2RpcG9kaTpkb2NuYW1lPSJ3aXRob3V0T3JkZXJieS5zdmciCiAgIGlua3NjYXBlOnZlcnNpb249IjEuMC4xIChjNDk3YjAzYywgMjAyMC0wOS0xMCkiPgogIDxtZXRhZGF0YQogICAgIGlkPSJtZXRhZGF0YTg0MSI+CiAgICA8cmRmOlJERj4KICAgICAgPGNjOldvcmsKICAgICAgICAgcmRmOmFib3V0PSIiPgogICAgICAgIDxkYzpmb3JtYXQ+aW1hZ2Uvc3ZnK3htbDwvZGM6Zm9ybWF0PgogICAgICAgIDxkYzp0eXBlCiAgICAgICAgICAgcmRmOnJlc291cmNlPSJodHRwOi8vcHVybC5vcmcvZGMvZGNtaXR5cGUvU3RpbGxJbWFnZSIgLz4KICAgICAgICA8ZGM6dGl0bGU+PC9kYzp0aXRsZT4KICAgICAgPC9jYzpXb3JrPgogICAgPC9yZGY6UkRGPgogIDwvbWV0YWRhdGE+CiAgPGRlZnMKICAgICBpZD0iZGVmczgzOSIgLz4KICA8c29kaXBvZGk6bmFtZWR2aWV3CiAgICAgcGFnZWNvbG9yPSIjZmZmZmZmIgogICAgIGJvcmRlcmNvbG9yPSIjNjY2NjY2IgogICAgIGJvcmRlcm9wYWNpdHk9IjEiCiAgICAgb2JqZWN0dG9sZXJhbmNlPSIxMCIKICAgICBncmlkdG9sZXJhbmNlPSIxMCIKICAgICBndWlkZXRvbGVyYW5jZT0iMTAiCiAgICAgaW5rc2NhcGU6cGFnZW9wYWNpdHk9IjAiCiAgICAgaW5rc2NhcGU6cGFnZXNoYWRvdz0iMiIKICAgICBpbmtzY2FwZTp3aW5kb3ctd2lkdGg9IjE0MzYiCiAgICAgaW5rc2NhcGU6d2luZG93LWhlaWdodD0iNzk3IgogICAgIGlkPSJuYW1lZHZpZXc4MzciCiAgICAgc2hvd2dyaWQ9ImZhbHNlIgogICAgIGlua3NjYXBlOnpvb209IjQwLjMxMjcyMSIKICAgICBpbmtzY2FwZTpjeD0iMC4zMjcwODA1NyIKICAgICBpbmtzY2FwZTpjeT0iNi41NDY3NDk2IgogICAgIGlua3NjYXBlOndpbmRvdy14PSIwIgogICAgIGlua3NjYXBlOndpbmRvdy15PSIyMyIKICAgICBpbmtzY2FwZTp3aW5kb3ctbWF4aW1pemVkPSIwIgogICAgIGlua3NjYXBlOmN1cnJlbnQtbGF5ZXI9InN2ZzgzNSIgLz4KICA8cGF0aAogICAgIG9wYWNpdHk9IjAuNTUzMzYyIgogICAgIGZpbGwtcnVsZT0iZXZlbm9kZCIKICAgICBjbGlwLXJ1bGU9ImV2ZW5vZGQiCiAgICAgZD0iTSAwLDUgNC4xMTE4MDYsMCA3Ljk5OTk5Niw1IFoiCiAgICAgZmlsbD0id2hpdGUiCiAgICAgaWQ9InBhdGg4MzMiCiAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MC41NTAwMDAwMSIgLz4KICA8ZwogICAgIHN0eWxlPSJmaWxsOiNmZmZmZmY7ZmlsbC1vcGFjaXR5OjAuNTUwMDAwMDEiCiAgICAgaWQ9Imc4NTQiCiAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTAuMTUwMDIyLDcpIj4KICAgIDxwYXRoCiAgICAgICBvcGFjaXR5PSIwLjU1MzM2MiIKICAgICAgIGZpbGwtcnVsZT0iZXZlbm9kZCIKICAgICAgIGNsaXAtcnVsZT0iZXZlbm9kZCIKICAgICAgIGQ9Im0gOC4xNTAwMiwwLjUgLTQuMTExOCw1IC0zLjg4ODE5NiwtNSB6IgogICAgICAgZmlsbD0iI2ZmZmZmZiIKICAgICAgIGlkPSJwYXRoODQzIgogICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MC41NTAwMDAwMSIgLz4KICA8L2c+Cjwvc3ZnPgo=');
  background-repeat: no-repeat;
  background-position: center;
}

.ui-list-table thead th .orderByAsc {
  width: 7px;
  height: 14px;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxzdmcKICAgeG1sbnM6ZGM9Imh0dHA6Ly9wdXJsLm9yZy9kYy9lbGVtZW50cy8xLjEvIgogICB4bWxuczpjYz0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjIgogICB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiCiAgIHhtbG5zOnN2Zz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciCiAgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgeG1sbnM6c29kaXBvZGk9Imh0dHA6Ly9zb2RpcG9kaS5zb3VyY2Vmb3JnZS5uZXQvRFREL3NvZGlwb2RpLTAuZHRkIgogICB4bWxuczppbmtzY2FwZT0iaHR0cDovL3d3dy5pbmtzY2FwZS5vcmcvbmFtZXNwYWNlcy9pbmtzY2FwZSIKICAgd2lkdGg9IjgiCiAgIGhlaWdodD0iMTIuNSIKICAgdmlld0JveD0iMCAwIDggMTIuNSIKICAgZmlsbD0ibm9uZSIKICAgdmVyc2lvbj0iMS4xIgogICBpZD0ic3ZnODM1IgogICBzb2RpcG9kaTpkb2NuYW1lPSJPcmRlcmJ5QXNjLnN2ZyIKICAgaW5rc2NhcGU6dmVyc2lvbj0iMS4wLjEgKGM0OTdiMDNjLCAyMDIwLTA5LTEwKSI+CiAgPG1ldGFkYXRhCiAgICAgaWQ9Im1ldGFkYXRhODQxIj4KICAgIDxyZGY6UkRGPgogICAgICA8Y2M6V29yawogICAgICAgICByZGY6YWJvdXQ9IiI+CiAgICAgICAgPGRjOmZvcm1hdD5pbWFnZS9zdmcreG1sPC9kYzpmb3JtYXQ+CiAgICAgICAgPGRjOnR5cGUKICAgICAgICAgICByZGY6cmVzb3VyY2U9Imh0dHA6Ly9wdXJsLm9yZy9kYy9kY21pdHlwZS9TdGlsbEltYWdlIiAvPgogICAgICAgIDxkYzp0aXRsZT48L2RjOnRpdGxlPgogICAgICA8L2NjOldvcms+CiAgICA8L3JkZjpSREY+CiAgPC9tZXRhZGF0YT4KICA8ZGVmcwogICAgIGlkPSJkZWZzODM5IiAvPgogIDxzb2RpcG9kaTpuYW1lZHZpZXcKICAgICBwYWdlY29sb3I9IiNmZmZmZmYiCiAgICAgYm9yZGVyY29sb3I9IiM2NjY2NjYiCiAgICAgYm9yZGVyb3BhY2l0eT0iMSIKICAgICBvYmplY3R0b2xlcmFuY2U9IjEwIgogICAgIGdyaWR0b2xlcmFuY2U9IjEwIgogICAgIGd1aWRldG9sZXJhbmNlPSIxMCIKICAgICBpbmtzY2FwZTpwYWdlb3BhY2l0eT0iMCIKICAgICBpbmtzY2FwZTpwYWdlc2hhZG93PSIyIgogICAgIGlua3NjYXBlOndpbmRvdy13aWR0aD0iMTQzNiIKICAgICBpbmtzY2FwZTp3aW5kb3ctaGVpZ2h0PSI3OTciCiAgICAgaWQ9Im5hbWVkdmlldzgzNyIKICAgICBzaG93Z3JpZD0iZmFsc2UiCiAgICAgaW5rc2NhcGU6em9vbT0iNDAuMzEyNzIxIgogICAgIGlua3NjYXBlOmN4PSIwLjMyNzA4MDU3IgogICAgIGlua3NjYXBlOmN5PSI2LjU0Njc0OTYiCiAgICAgaW5rc2NhcGU6d2luZG93LXg9IjAiCiAgICAgaW5rc2NhcGU6d2luZG93LXk9IjIzIgogICAgIGlua3NjYXBlOndpbmRvdy1tYXhpbWl6ZWQ9IjAiCiAgICAgaW5rc2NhcGU6Y3VycmVudC1sYXllcj0ic3ZnODM1IiAvPgogIDxwYXRoCiAgICAgb3BhY2l0eT0iMC41NTMzNjIiCiAgICAgZmlsbC1ydWxlPSJldmVub2RkIgogICAgIGNsaXAtcnVsZT0iZXZlbm9kZCIKICAgICBkPSJNIDAsNSA0LjExMTgwNiwwIDcuOTk5OTk2LDUgWiIKICAgICBmaWxsPSJ3aGl0ZSIKICAgICBpZD0icGF0aDgzMyIKICAgICBzdHlsZT0iZmlsbDojZmZmZmZmO2ZpbGwtb3BhY2l0eToxIiAvPgogIDxnCiAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MC41NTAwMDAwMSIKICAgICBpZD0iZzg1NCIKICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMC4xNTAwMjIsNykiPgogICAgPHBhdGgKICAgICAgIG9wYWNpdHk9IjAuNTUzMzYyIgogICAgICAgZmlsbC1ydWxlPSJldmVub2RkIgogICAgICAgY2xpcC1ydWxlPSJldmVub2RkIgogICAgICAgZD0ibSA4LjE1MDAyLDAuNSAtNC4xMTE4LDUgLTMuODg4MTk2LC01IHoiCiAgICAgICBmaWxsPSIjZmZmZmZmIgogICAgICAgaWQ9InBhdGg4NDMiCiAgICAgICBzdHlsZT0iZmlsbDojZmZmZmZmO2ZpbGwtb3BhY2l0eTowLjU1MDAwMDAxIiAvPgogIDwvZz4KPC9zdmc+Cg==');
  background-repeat: no-repeat;
  background-position: center;
}

.ui-list-table thead th .orderByDesc {
  width: 7px;
  height: 14px;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxzdmcKICAgeG1sbnM6ZGM9Imh0dHA6Ly9wdXJsLm9yZy9kYy9lbGVtZW50cy8xLjEvIgogICB4bWxuczpjYz0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjIgogICB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiCiAgIHhtbG5zOnN2Zz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciCiAgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgeG1sbnM6c29kaXBvZGk9Imh0dHA6Ly9zb2RpcG9kaS5zb3VyY2Vmb3JnZS5uZXQvRFREL3NvZGlwb2RpLTAuZHRkIgogICB4bWxuczppbmtzY2FwZT0iaHR0cDovL3d3dy5pbmtzY2FwZS5vcmcvbmFtZXNwYWNlcy9pbmtzY2FwZSIKICAgd2lkdGg9IjgiCiAgIGhlaWdodD0iMTIuNSIKICAgdmlld0JveD0iMCAwIDggMTIuNSIKICAgZmlsbD0ibm9uZSIKICAgdmVyc2lvbj0iMS4xIgogICBpZD0ic3ZnODM1IgogICBzb2RpcG9kaTpkb2NuYW1lPSJPcmRlcmJ5RGVzYy5zdmciCiAgIGlua3NjYXBlOnZlcnNpb249IjEuMC4xIChjNDk3YjAzYywgMjAyMC0wOS0xMCkiPgogIDxtZXRhZGF0YQogICAgIGlkPSJtZXRhZGF0YTg0MSI+CiAgICA8cmRmOlJERj4KICAgICAgPGNjOldvcmsKICAgICAgICAgcmRmOmFib3V0PSIiPgogICAgICAgIDxkYzpmb3JtYXQ+aW1hZ2Uvc3ZnK3htbDwvZGM6Zm9ybWF0PgogICAgICAgIDxkYzp0eXBlCiAgICAgICAgICAgcmRmOnJlc291cmNlPSJodHRwOi8vcHVybC5vcmcvZGMvZGNtaXR5cGUvU3RpbGxJbWFnZSIgLz4KICAgICAgICA8ZGM6dGl0bGU+PC9kYzp0aXRsZT4KICAgICAgPC9jYzpXb3JrPgogICAgPC9yZGY6UkRGPgogIDwvbWV0YWRhdGE+CiAgPGRlZnMKICAgICBpZD0iZGVmczgzOSIgLz4KICA8c29kaXBvZGk6bmFtZWR2aWV3CiAgICAgcGFnZWNvbG9yPSIjZmZmZmZmIgogICAgIGJvcmRlcmNvbG9yPSIjNjY2NjY2IgogICAgIGJvcmRlcm9wYWNpdHk9IjEiCiAgICAgb2JqZWN0dG9sZXJhbmNlPSIxMCIKICAgICBncmlkdG9sZXJhbmNlPSIxMCIKICAgICBndWlkZXRvbGVyYW5jZT0iMTAiCiAgICAgaW5rc2NhcGU6cGFnZW9wYWNpdHk9IjAiCiAgICAgaW5rc2NhcGU6cGFnZXNoYWRvdz0iMiIKICAgICBpbmtzY2FwZTp3aW5kb3ctd2lkdGg9IjE0MzYiCiAgICAgaW5rc2NhcGU6d2luZG93LWhlaWdodD0iNzk3IgogICAgIGlkPSJuYW1lZHZpZXc4MzciCiAgICAgc2hvd2dyaWQ9ImZhbHNlIgogICAgIGlua3NjYXBlOnpvb209IjQwLjMxMjcyMSIKICAgICBpbmtzY2FwZTpjeD0iMC4zMjcwODA1NyIKICAgICBpbmtzY2FwZTpjeT0iNi41NDY3NDk2IgogICAgIGlua3NjYXBlOndpbmRvdy14PSIwIgogICAgIGlua3NjYXBlOndpbmRvdy15PSIyMyIKICAgICBpbmtzY2FwZTp3aW5kb3ctbWF4aW1pemVkPSIwIgogICAgIGlua3NjYXBlOmN1cnJlbnQtbGF5ZXI9InN2ZzgzNSIgLz4KICA8cGF0aAogICAgIG9wYWNpdHk9IjAuNTUzMzYyIgogICAgIGZpbGwtcnVsZT0iZXZlbm9kZCIKICAgICBjbGlwLXJ1bGU9ImV2ZW5vZGQiCiAgICAgZD0iTSAwLDUgNC4xMTE4MDYsMCA3Ljk5OTk5Niw1IFoiCiAgICAgZmlsbD0id2hpdGUiCiAgICAgaWQ9InBhdGg4MzMiCiAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MC41NTAwMDAwMSIgLz4KICA8ZwogICAgIHN0eWxlPSJmaWxsOiNmZmZmZmY7ZmlsbC1vcGFjaXR5OjEiCiAgICAgaWQ9Imc4NTQiCiAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTAuMTUwMDIyLDcpIj4KICAgIDxwYXRoCiAgICAgICBvcGFjaXR5PSIwLjU1MzM2MiIKICAgICAgIGZpbGwtcnVsZT0iZXZlbm9kZCIKICAgICAgIGNsaXAtcnVsZT0iZXZlbm9kZCIKICAgICAgIGQ9Im0gOC4xNTAwMiwwLjUgLTQuMTExOCw1IC0zLjg4ODE5NiwtNSB6IgogICAgICAgZmlsbD0iI2ZmZmZmZiIKICAgICAgIGlkPSJwYXRoODQzIgogICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MSIgLz4KICA8L2c+Cjwvc3ZnPgo=');
  background-repeat: no-repeat;
  background-position: center;
}

.ui-list-table tbody td {
  background: #ffffff;
  border: 1px solid #f2f2f2;
  box-sizing: border-box;
  padding: 8.5px 14px;
  font-size: 13px;
  vertical-align: middle;
}

.ui-list-table tbody td img {
  width: 100px;
}

.helper-message {
  text-align: center;
  font-size: 14px;
  margin: 10px;
}

.ui-list-table .redirect-field {
  color: #4e71c0;
  text-decoration: underline;
  cursor: pointer;
}

.ui-list-table .tr-accordion {
  cursor: pointer;
}

.ui-list-table .tr-accordion .plus {
  width: 14px;
  height: 14px;
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTMiIGhlaWdodD0iMTMiIHZpZXdCb3g9IjAgMCAxMyAxMyIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik04IDBINVY1SDBWOEg1VjEzSDhWOEgxM1Y1SDhWMFoiIGZpbGw9IiM5OTk5OTkiLz4KPC9zdmc+Cg==');
  background-repeat: no-repeat;
  background-position: center;
  float: left;
  cursor: pointer;
}

.ui-list-table .tr-accordion .minus {
  width: 14px;
  height: 14px;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxzdmcKICAgeG1sbnM6ZGM9Imh0dHA6Ly9wdXJsLm9yZy9kYy9lbGVtZW50cy8xLjEvIgogICB4bWxuczpjYz0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjIgogICB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiCiAgIHhtbG5zOnN2Zz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciCiAgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgeG1sbnM6c29kaXBvZGk9Imh0dHA6Ly9zb2RpcG9kaS5zb3VyY2Vmb3JnZS5uZXQvRFREL3NvZGlwb2RpLTAuZHRkIgogICB4bWxuczppbmtzY2FwZT0iaHR0cDovL3d3dy5pbmtzY2FwZS5vcmcvbmFtZXNwYWNlcy9pbmtzY2FwZSIKICAgd2lkdGg9IjEzIgogICBoZWlnaHQ9IjEzIgogICB2aWV3Qm94PSIwIDAgMTMgMTMiCiAgIGZpbGw9Im5vbmUiCiAgIHZlcnNpb249IjEuMSIKICAgaWQ9InN2ZzEyIgogICBzb2RpcG9kaTpkb2NuYW1lPSJNaW51cy5zdmciCiAgIGlua3NjYXBlOnZlcnNpb249IjEuMC4xIChjNDk3YjAzYywgMjAyMC0wOS0xMCkiPgogIDxtZXRhZGF0YQogICAgIGlkPSJtZXRhZGF0YTE4Ij4KICAgIDxyZGY6UkRGPgogICAgICA8Y2M6V29yawogICAgICAgICByZGY6YWJvdXQ9IiI+CiAgICAgICAgPGRjOmZvcm1hdD5pbWFnZS9zdmcreG1sPC9kYzpmb3JtYXQ+CiAgICAgICAgPGRjOnR5cGUKICAgICAgICAgICByZGY6cmVzb3VyY2U9Imh0dHA6Ly9wdXJsLm9yZy9kYy9kY21pdHlwZS9TdGlsbEltYWdlIiAvPgogICAgICAgIDxkYzp0aXRsZT48L2RjOnRpdGxlPgogICAgICA8L2NjOldvcms+CiAgICA8L3JkZjpSREY+CiAgPC9tZXRhZGF0YT4KICA8ZGVmcwogICAgIGlkPSJkZWZzMTYiIC8+CiAgPHNvZGlwb2RpOm5hbWVkdmlldwogICAgIHBhZ2Vjb2xvcj0iI2ZmZmZmZiIKICAgICBib3JkZXJjb2xvcj0iIzY2NjY2NiIKICAgICBib3JkZXJvcGFjaXR5PSIxIgogICAgIG9iamVjdHRvbGVyYW5jZT0iMTAiCiAgICAgZ3JpZHRvbGVyYW5jZT0iMTAiCiAgICAgZ3VpZGV0b2xlcmFuY2U9IjEwIgogICAgIGlua3NjYXBlOnBhZ2VvcGFjaXR5PSIwIgogICAgIGlua3NjYXBlOnBhZ2VzaGFkb3c9IjIiCiAgICAgaW5rc2NhcGU6d2luZG93LXdpZHRoPSIxOTIwIgogICAgIGlua3NjYXBlOndpbmRvdy1oZWlnaHQ9IjEwMjciCiAgICAgaWQ9Im5hbWVkdmlldzE0IgogICAgIHNob3dncmlkPSJmYWxzZSIKICAgICBpbmtzY2FwZTp6b29tPSI1Ni41Mzg0NjIiCiAgICAgaW5rc2NhcGU6Y3g9IjYuNSIKICAgICBpbmtzY2FwZTpjeT0iMy4wNTEwMjA0IgogICAgIGlua3NjYXBlOndpbmRvdy14PSIwIgogICAgIGlua3NjYXBlOndpbmRvdy15PSIyNSIKICAgICBpbmtzY2FwZTp3aW5kb3ctbWF4aW1pemVkPSIxIgogICAgIGlua3NjYXBlOmN1cnJlbnQtbGF5ZXI9InN2ZzEyIiAvPgogIDxwYXRoCiAgICAgZmlsbC1ydWxlPSJldmVub2RkIgogICAgIGNsaXAtcnVsZT0iZXZlbm9kZCIKICAgICBkPSJNIDUsNSBIIDAgdiAzIGggNSAzIDUgViA1IEMgMTMsNSA5LjY2NjY2NjcsNSA4LDUgNyw1IDUsNSA1LDUgWiIKICAgICBmaWxsPSIjOTk5OTk5IgogICAgIGlkPSJwYXRoMTAiCiAgICAgc29kaXBvZGk6bm9kZXR5cGVzPSJjY2NjY2Njc2MiIC8+Cjwvc3ZnPgo=');
  background-repeat: no-repeat;
  background-position: center;
  float: left;
  cursor: pointer;
}

.ui-list-table .tr-selected td {
  background-color: rgb(250, 251, 227) !important;
}

.ui-list-table .tr-accordion-details td {
  padding: 0px;
}

.ui-list-table .tr-accordion-details td div {
  max-height: 0;
  padding: 0 20px;
  box-sizing: border-box;
  overflow: hidden;
  transition: max-height 0.3s, padding 0.5s;
}

.ui-list-table .tr-accordion-details td div table thead th {
  background-color: rgb(250, 251, 227) !important;
  color: black;
}

.ui-list-table .tr-accordion-details-show td div {
  max-height: 2000px;
  padding: 18px 31px 45px 36px;
  transition: max-height 0.3s, padding 0.3s;
  background-color: rgb(239, 239, 239);
  box-shadow: inset 1px 0px 10px rgba(0, 0, 0, 0.1);
}

.ui-list-table .tr-accordion-details-show td div table thead th {
  background-color: rgb(250, 251, 227) !important;
  color: black;
}

.ui-list-table .tr-accordion-details-show td div table tbody tr td {
  padding: 7px;
}

.tooltip {
  position: relative;
  display: inline-block;
}

/* Tooltip text */
.tooltip .tooltiptext {
  visibility: hidden;
  width: 160px;
  background-color: white;
  color: #333333;
  text-align: center;
  padding: 5px 0;
  border-radius: 6px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  /* Position the tooltip text - see examples below! */
  position: absolute;
  z-index: 1;
  right: 20px;
  bottom: 20px;
}

/* Show the tooltip text when you mouse over the tooltip container */
.tooltip:hover .tooltiptext {
  visibility: visible;
}

.edit-icon {
  position: absolute;
  width: 12px;
  height: 14px;
  cursor: pointer;
  margin-left: 10px;
  background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA0AAAANCAYAAABy6+R8AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAACDSURBVHgBlZKBDYAgDAQrcQBHYAQZwQmcgckYxRF0AxmBEWxJa4ia8H7yQEkuX1KIAKWUJrHVDgF429g7n73cDQDgFZJzZgcHJKzshV3EMcYydoBZTQrmz/YegOnghGCFQwBNuTUggLzjBf0B2vZgoCbpwE4UaJNgoCbJYt+DdHjU0QXXcT/ULxTUEAAAAABJRU5ErkJggg==');
}

.tooltip {
  position: relative;
  display: inline-block;
}

.tooltip-edit:before {
  content: '';
  top: -9px;
  left: 170px;
  position: absolute;
  width: 15px;
  height: 15px;
  background-color: #fff;
  transform: rotate(45deg);
  z-index: 2;
  border-top: 1px solid #ccc;
  border-left: 1px solid #ccc;
}

.tooltip-edit {
  position: relative;
  z-index: 1;
  width: 400px;
  background: #ffffff;
  border-radius: 4px;
  border: 0.7px solid #e1e1e1;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  padding: 20px 15px;
  margin-top: 12px;
}

.tooltip .tooltip-information-container {
  transition-duration: 1s;
  max-height: 400px;
  height: auto;
  overflow: hidden;
  position: absolute;
  left: -140%;
}

.tooltip .tooltip-information-container.hidden {
  max-height: 0px;
}

.container {
  display: flex;
  max-width: 1200px;
}

.column-left {
  width: 180px;
  flex: 0 0 120px;
  margin: 10px 0px;
}

.column-left label {
  margin: 0 10px 0px 90px;
}

.column-right {
  width: 230px;
  flex: 0 0 230px;
  margin: 10px 0px;
}

.column-right input {
  height: 25px;
}

.message-info {
  display: flex;
  width: 100%;
  margin: 5px;
  align-items: center;
}

.message-info svg {
  margin-right: 10px;
}

.container-buttons {
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.container-buttons .submit-button {
  width: auto;
  height: 29px;
  font-family: 'Helvetica Neue';
  font-style: normal;
  cursor: pointer;
  font-weight: 400;
  font-size: 14px;
  line-height: 21px;
  color: #ffffff;
  background: #4e71c0;
  box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.25), inset 1px 1px 2px rgba(255, 255, 255, 0.2);
  border-radius: 1px;
}

.container-buttons button:disabled {
  background-color: #849eda;
  cursor: default;
  color: #ffffff;
}

.container-buttons .cancel-button {
  width: 72px;
  height: 29px;
  font-family: 'Helvetica Neue';
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 21px;
  background: #f2f2f2;
  border: 0.5px solid rgba(102, 102, 102, 0.4);
  border-radius: 1px;
  color: #666666;
  margin-left: 10px;
  cursor: pointer;
}

.label-info {
  background: #F2F2F2;
  border: 0.3px solid #767676;
  border-radius: 20px;
  text-align: center;
  line-height: 16px;
  color: #767676;
  width: 90px;
  padding: 2px;
  font-size: 13px;
  cursor: pointer;
}

.tooltip-info {
  position: relative;
}

.tooltip-content {
  position: absolute;
  width: 300px;
  height: 160px;
  background: #ffffff;
  border-radius: 4px;
  border: 0.7px solid #e1e1e1;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
  z-index: 20;
  margin-top: 10px;
  right: -35px;
  padding: 10px 20px;
  font-weight: 400;
  font-size: 13px;
  line-height: 16px;
}

.tooltip-content:before {
  content: '';
  top: -9px;
  left: 50%;
  position: absolute;
  width: 15px;
  height: 15px;
  background-color: #fff;
  transform: rotate(45deg);
  z-index: 2;
  border-top: 1px solid #ccc;
  border-left: 1px solid #ccc;
}

.date-info {
  opacity: 0.5;
}