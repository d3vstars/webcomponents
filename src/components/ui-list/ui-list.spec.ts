import { newSpecPage } from '@stencil/core/testing';
import { UIList } from './ui-list';

describe('ui-list', () => {
  it('builds', async () => {
    expect(new UIList()).toBeTruthy();
  });

  it('renders', async () => {
    const page = await newSpecPage({
      components: [UIList],
      html: `<ui-list></ui-list>`
    });

    expect(page.root).toEqualHtml(`<ui-list><mock:shadow-root>
    <div class="ui-list-table-content"><table class="ui-list-table"><thead></thead><tbody></tbody></table></div>
    </ui-list>`);
  });

  it('renders with data', async () => {
    const page = await newSpecPage({
      components: [UIList],
      html: '<div></div>'
    });

    let cmp = page.doc.createElement('ui-list');
    cmp.dataTable = [{ name: 'John' }];

    cmp.headers = [
      {
        // @ts-ignore
        key: 'name',
        label: 'Name',
        searchable: false,
        render: (nameAttribute: string) => {
          return nameAttribute;
        }
      }
    ];

    page.root.appendChild(cmp);
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });
});
