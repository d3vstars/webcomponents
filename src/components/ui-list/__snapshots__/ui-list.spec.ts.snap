// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-list renders with data 1`] = `
<ui-list>
  <div class="ui-list-table-content">
    <table class="ui-list-table">
      <thead>
        <th>
          Name
        </th>
      </thead>
      <tbody>
        <tr>
          <td class="name">
            John
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</ui-list>
`;
